src/Features/Lsif/Generator/Program.cs

```diff
@@ -51,12 +52,16 @@ private static async Task GenerateAsync(FileInfo? solution, FileInfo? compilerIn
 
             try
             {
-                // Exactly one of "solution" or "compilerInvocation" should be specified
-                if (solution != null && compilerInvocation == null)
+                // Exactly one of "solution", or "project" or "compilerInvocation" should be specified
+                if (solution != null && project == null && compilerInvocation == null)
                 {
                     await GenerateFromSolutionAsync(solution, lsifWriter, logFile);
                 }
-                else if (compilerInvocation != null && solution == null)
+                else if (solution == null && project != null && compilerInvocation == null)
+                {
+                    await GenerateFromProjectAsync(project, lsifWriter, logFile);
+                }
+                else if (solution == null && project == null && compilerInvocation != null)
                 {
                     await GenerateFromCompilerInvocationAsync(compilerInvocation, lsifWriter, logFile);
                 }
```

Comment:
The message in the "else" should be updated.

Reply:
done.